cmd/crio/kubensenter/kubensenter.go

```diff
@@ -0,0 +1,33 @@
+//go:build linux && cgo
+
+package kubensenter
+
+/*
+#cgo CFLAGS: -Wall -D_GNU_SOURCE
+
+#include "kubensenter.h"
+
+#define ERR_LIMIT 128
+
+char* nsenterConfig = "";
+int nsenterResult = -1;
+char _errorMessage[ERR_LIMIT];
+char* errorMessage = _errorMessage; // Use an intermediary because C.GoString can't work on an array
+
+void __attribute__((constructor)) crio_nsenter_init(int argc, char **argv, char *envp[]) {
+	errorMessage[0] = '\0';
+	nsenterConfig = getKubeNsMnt();
+	nsenterResult = joinMountNamespace(nsenterConfig, errorMessage, ERR_LIMIT);
+}
+*/
+import "C"
+
+import "errors"
+
+func Status() (string, error) {
```

Comment:
the name Status doesn't imply that it is also joining the namespace. Maybe "CheckAndEnter" or something?